src/ring_theory/polynomial/hermite.lean

```diff
@@ -124,5 +137,72 @@ begin
 end
 
 end coeff
+end polynomial
+
+/-! ### Lemmas about `polynomial.hermite_gauss` -/
+
+section gaussian
+
+/-- The real Gaussian function -/
+def gaussian : ℝ → ℝ := λ x, real.exp (-(x^2 / 2))
+
+lemma inv_gaussian_eq : gaussian⁻¹ = λ x, real.exp (x^2 / 2) :=
+by { ext, simp [gaussian, real.exp_neg] }
+
+lemma inv_gaussian_mul_gaussian (x : ℝ) : gaussian⁻¹ x * gaussian x = 1 :=
+by rw [inv_gaussian_eq, gaussian, ← real.exp_add, add_neg_self, real.exp_zero]
+
+lemma deriv_gaussian (x : ℝ) : deriv gaussian x = -x * gaussian x :=
+by simp [gaussian, mul_comm]
+
+lemma deriv_inv_gaussian (x : ℝ) : deriv gaussian⁻¹ x = x * gaussian⁻¹ x :=
+by simp [inv_gaussian_eq, mul_comm]
+
+lemma cont_diff_gaussian : cont_diff ℝ ⊤ gaussian :=
+((cont_diff_id.pow 2).div_const 2).neg.exp
+
+lemma cont_diff.iterated_deriv :
+∀ (n : ℕ) (f : ℝ → ℝ) (hf : cont_diff ℝ ⊤ f), cont_diff ℝ ⊤ (deriv^[n] f)
+| 0     f hf := hf
+| (n+1) f hf := cont_diff.iterated_deriv n (deriv f) (cont_diff_top_iff_deriv.mp hf).2
+
+/-- The Gaussian form of `hermite n` -/
+def hermite_gauss (n : ℕ) : ℝ → ℝ :=
+λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)
+
+lemma hermite_gauss_def (n : ℕ) :
+hermite_gauss n = λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x) := rfl
+
+lemma hermite_gauss_succ (n : ℕ) : hermite_gauss (n+1)
+= id * (hermite_gauss n) - deriv (hermite_gauss n):=
+begin
+  ext,
+  simp only [hermite_gauss, function.iterate_succ', function.comp_app,
+             id.def, pi.mul_apply, pi.sub_apply, pow_succ],
+  rw [deriv_mul, deriv_const_mul, deriv_inv_gaussian],
+  ring,
+  { simp [inv_gaussian_eq] },
+  { simp [inv_gaussian_eq] },
+  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterated_deriv _ _ cont_diff_gaussian)).1 }
+end
+
+namespace polynomial
+
+lemma hermite_eq_gauss (n : ℕ) :
+(λ x, eval x (map (algebra_map ℤ ℝ) (hermite n))) =
```

Comment:
I think this is `aeval`?

Reply:
Hmm, there doesn't seem to be an analog of `polynomial.deriv : deriv (λx, p.eval x) x = p.derivative.eval x :=` for `aeval`. This breaks the `simp` on line 199.

Reply:
There should be a lemma that rewrites `aeval` into `eval _ (map _ _)`.

Another good answer is "that lemma should be added then"!

Reply:
Ah, found it: `eval₂_eq_eval_map`. Both `aeval` and `eval` are defined in terms of `eval₂`, but the simp lemmas actually aim for `eval`.